crates/bpf-common/src/program.rs

```diff
@@ -333,16 +346,18 @@ impl Program {
         map_name: &str,
         sender: impl BpfSender<T>,
     ) -> Result<(), ProgramError> {
+        if self.used_maps.contains(map_name) {
+            return Err(ProgramError::MapAlreadyUsed(map_name.to_string()));
+        };
+
         let map_resource = self
             .bpf
             .take_map(map_name)
             .ok_or_else(|| ProgramError::MapNotFound(map_name.to_string()))?;
+
+        self.used_maps.insert(map_name.to_string());
+
```

Comment:
We could make a shared method since the code is duplicated: 

```rust
fn take_map(&mut self, map_name: str) -> _ {
        if self.used_maps.contains(map_name) {
            return Err(ProgramError::MapAlreadyUsed(map_name.to_string()));
        };

        let map_resource = self
            .bpf
            .take_map(map_name)
            .ok_or_else(|| ProgramError::MapNotFound(map_name.to_string()))?;

        self.used_maps.insert(map_name.to_string());
        Ok(map_resource)
}
```